src/data-model-providers/codegen/CodegenDataModelProvider_Write.cpp

```diff
@@ -94,14 +94,6 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::WriteAttribute(const Dat
     ChipLogDetail(DataManagement, "Writing attribute: Cluster=" ChipLogFormatMEI " Endpoint=0x%x AttributeId=" ChipLogFormatMEI,
                   ChipLogValueMEI(request.path.mClusterId), request.path.mEndpointId, ChipLogValueMEI(request.path.mAttributeId));
 
-    // TODO: ordering is to check writability/existence BEFORE ACL and this seems wrong, however
-    //       existing unit tests (TC_AcessChecker.py) validate that we get UnsupportedWrite instead of UnsupportedAccess
-    //
-    //       This should likely be fixed in spec (probably already fixed by
-    //       https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9024)
-    //       and tests and implementation
-    //
-    //       Open issue that needs fixing: https://github.com/project-chip/connectedhomeip/issues/33735
     auto metadata = Ember::FindAttributeMetadata(request.path);
 
     // Explicit failure in finding a suitable metadata
```

Comment:
Given our caller is now responsible for doing the existence check, can we just VerifyOrDie that the metadata exists?

Reply:
I think the win of a `VerifyOrDie` vs existing logic which is 2 compares is minimal. I can also unit test this and cannot unit test VerifyOrDie.

So while correctness-wise we could, I think existing path is ok: we return errors if prerequisites are not met and process if prerequisites are ok.

We can re-consider as a followup maybe, after we discuss what approaches are best. If cost is minimal, I prefer erroring out to verifyordie.

Reply:
OK, that works.  Just wanted to check whether we could simplify here....